attr: Add NambooripadPartialOrder & Leq methods

ChristopherRussell opened the pull request:
The Nambooripad partial order is a generalization of the natural partial
order for inverse semigroups to the class of regular semigroups. These
functions are modelled on the fallback methods for NaturalPartialOrder
and NaturalLeqInverseSemigroup. As I do not have a current use for this,
I haven't specialized the methods for any particular representations of
regular semigroups.

The order is defined on a regular semigroup by is x <= y if the respective
R-classes satisfy R_x <= R_y, and there is an idempotent e in R_x such
that x = e * y.

james-d-mitchell commented on doc/attr.xml:
```diff
@@ -1233,3 +1233,74 @@ gap> IndecomposableElements(B);
+    <A>S</A> is defined by <C>s</C><M>\leq</M><C>t</C> if the principal right
+    ideal of <C>S</C> generated by <C>s</C> is contained in the principal right
+    ideal of <C>S</C> generated by <C>t</C> and there is an idempotent
+    <C>e</C> in the R-class of <C>s</C> such that <C>s</C><M>=</M><C>et</C>. The
```
This should be `&R;-class` so that it is displayed correctly.

james-d-mitchell commented on doc/attr.xml:
```diff
@@ -1233,3 +1233,74 @@ gap> IndecomposableElements(B);
+<Attr Name="NambooripadPartialOrder" Arg="S"/>
+  <Returns>The Nambooripad partial order on an regular semigroup.</Returns>
+  <Description>
+    The <E>Nambooripad partial order</E> <M>\leq</M> on an regular semigroup
```
"an regular" -> "a regular"

james-d-mitchell commented on doc/attr.xml:
```diff
@@ -1233,3 +1233,74 @@ gap> IndecomposableElements(B);
+    ideal of <C>S</C> generated by <C>t</C> and there is an idempotent
+    <C>e</C> in the R-class of <C>s</C> such that <C>s</C><M>=</M><C>et</C>. The
+    Nambooripad partial order coincides with the natural partial order when
+    considering inverse semigroups.
```
Add a reference to `NaturalPartialOrder` here?

james-d-mitchell commented on gap/attributes/attr.gi:
```diff
@@ -1080,3 +1080,57 @@ function(S)
+                  "the argument is not a finite semigroup,");
+  elif not IsRegularSemigroup(S) then
+    ErrorNoReturn("Semigroups: NambooripadPartialOrder: usage,\n",
+                  "the argument is not an inverse semigroup,");
```
"an inverse" -> "a regular"

james-d-mitchell commented on gap/attributes/attr.gi:
```diff
@@ -1080,3 +1080,57 @@ function(S)
+
+  if not IsRegularSemigroup(S) then
+    ErrorNoReturn("Semigroups: NambooripadLeqRegularSemigroup: usage,\n",
+                  "the argument is not an inverse semigroup,");
```
"an inverse" -> "a regular"

james-d-mitchell commented on gap/attributes/attr.gi:
```diff
@@ -1080,3 +1080,57 @@ function(S)
+    ErrorNoReturn("Semigroups: NambooripadLeqRegularSemigroup: usage,\n",
+                  "the argument is not a finite semigroup,");
+  fi;
+
```
Remove this blank line and replace the next `if not IsRegular...` by `elif not IsRegular...`

james-d-mitchell commented:
One other comment would be to perhaps make an issue suggesting that specialised methods for `Nambooripad*` could be installed for special types of semigroups.

ChristopherRussell commented:
Thanks @james-d-mitchell. I believe I have now fixed all these typos, added appropriate references to `NaturalPartialOrder` and delegated calculations to `NaturalPartialOrder` in the case of an inverse semigroup.

james-d-mitchell commented on doc/attr.xml:
```diff
@@ -1233,3 +1233,74 @@ gap> IndecomposableElements(B);
+
+    <C>NambooripadPartialOrder</C> returns the Nambooripad partial order on the
+    regular semigroup <A>S</A> as a list of sets of positive integers where
+    entry <C>i</C> in <C>NaturalPartialOrder(<A>S</A>)</C> is the set of
```
`NaturalPartialOrder` -> `NambooripadPartialOrder`
